org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/AssignToVariableAssistProposal.java

```diff
@@ -255,13 +257,18 @@ private ASTRewrite doAddField(ASTRewrite rewrite, ASTNode nodeToAssign, ITypeBin
 			return null;
 		}
 
-		IJavaProject project= getCompilationUnit().getJavaProject();
+		ICompilationUnit compilationUnit = getCompilationUnit();
+		IJavaProject project= compilationUnit.getJavaProject();
 		boolean isAnonymous= newTypeDecl.getNodeType() == ASTNode.ANONYMOUS_CLASS_DECLARATION;
 		boolean isStatic= Modifier.isStatic(bodyDecl.getModifiers()) && !isAnonymous;
 		int modifiers= Modifier.PRIVATE;
 		if (isStatic) {
 			modifiers |= Modifier.STATIC;
 		}
+		Preferences prefs = PreferenceManager.getPrefs(compilationUnit.getResource());
+		if (isParamToField && prefs.getCodeGenerationNewFieldsFinal()) {
+			modifiers |= Modifier.FINAL;
+		}
```

Comment:
Is this also used in other contexts where adding `final` wouldn't make sense?

Reply:
Looking at Eclipse's `final` cleanup (which was recently added as a cleanup action in https://github.com/eclipse/eclipse.jdt.ls/pull/2350/files#diff-c1fc3fbd7efda0e6e347dfc818586804aa68547c02b7c9abd75794e7a7481fadR46 .

![eclipse-cleanup-final-prefs](https://user-images.githubusercontent.com/1417342/207168792-b72e9de7-926c-4a8a-afdf-6ebeec826855.png)

So you could apply it to local variables, and even parameters. In this file I do see `doAddparam`, `doAddLocal`, but not sure how easy it is to accomplish that.

Another place this might go is in https://github.com/eclipse/eclipse.jdt.ls/blob/3400ec33d2b05e6ee639d90e2d3e1ed509c4b495/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/NewVariableCorrectionProposal.java#L391-L393